src/features/Account/RegisterForm.tsx

```diff
@@ -6,53 +7,60 @@ import { HiAtSymbol, HiFingerPrint, HiUser } from 'react-icons/hi';
 import { InputAdornment } from '@mui/material';
 import { useFormik } from 'formik';
 import { registerValidate } from './validate';
+import { oauthButtonStyle, oauthButtonProps } from './loginTheme';
+import { signIn } from 'next-auth/react';
+import { useRouter } from 'next/router';
+
 
 const Register = () => {
   const formik = useFormik({
     initialValues: {
-      username: '',
       email: '',
       password: '',
       cpassword: '',
+      userType: '',
     },
     validate: registerValidate,
     onSubmit: onSubmit
   });
 
+  const router = useRouter();
+  const callbackUrl = (router.query?.callbackUrl as string) || '/'
+
   async function onSubmit(values) {
-    console.log(values);
+    const options = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(values)
+    }
+
+    await fetch(`${callbackUrl}api/auth/signup`, options)
+      .then(res => res.json())
+      .then(data => {
+        console.log(data);
+        if (data.error) {
+          console.log(data.error);
+        }
+        if (data) {
+          router.push('/login');
```

Comment:
push `callbackUrl` instead of `/login`

Reply:
This takes us to the home page instead of the login page. Do we want that or should it route back to /login?

Reply:
@grace-fillmore

Reply:
It should take the user to whatever page they came from, which is sometimes the home page. Does the url not look like `login?callbackUrl={route}` when coming to the login page from a protected route?

Reply:
And even if it only sends them to home, I would prefer that after being logged in than staying on the login page with only an icon in the corner to indicate that something changed

Reply:
Are we talking about login or registration? This is the callback for after a user submits the register form successfully.